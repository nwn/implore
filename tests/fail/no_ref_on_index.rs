use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op]
fn index(x: X, y: u32) -> &X {
    &X
}

fn main() {}
