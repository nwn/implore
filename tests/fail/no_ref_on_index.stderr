error: the first operand of `index` must be a reference
 --> $DIR/no_ref_on_index.rs:7:13
  |
7 | fn index(x: X, y: u32) -> &X {
  |             ^
